dspy/predict/predict.py

```diff
@@ -15,13 +15,21 @@ def __init__(self, signature, **config):
         self.reset()
 
     def reset(self):
-        self.backend = None
+        if dspy.settings.get("backend", None) is not None:
```

Comment:
Is `backend` a new magical keyword? :D
https://github.com/stanfordnlp/dspy/issues/658